streams/src/main/java/org/apache/kafka/streams/state/internals/metrics/RocksDBMetricsRecorder.java

```diff
@@ -71,20 +70,15 @@ public TaskId taskId() {
         return taskId;
     }
 
+    public void init(final StreamsMetricsImpl streamsMetrics,
```

Comment:
nit: maybe add a comment that this function is idempotent?

Also is it ever possible that this function is triggered multi-times for different taskId / streamsMetrics objects? If not should we throw an IllegalStateException if the taskId / streamsMetrics are not null and are different from the passed in values.

Reply:
Good point! I added checks for the parameters.